decred/decred/dcr/txscript.py

```diff
@@ -2503,71 +2620,98 @@ def extractPkScriptAddrs(version, pkScript, chainParams):
             pk = Curve.parsePubKey(encodedPK)
-            addrs.append(
-                crypto.AddressSecpPubKey(pk.serializeCompressed(), chainParams)
-            )
+            addrs.append(crypto.AddressSecpPubKey(pk.serializeCompressed(), netParams))
         return MultiSigTy, addrs, details.requiredSigs
 
     # Check for stake submission script.  Only stake-submission-tagged
     # pay-to-pubkey-hash and pay-to-script-hash are allowed.
     pkHash = extractStakePubKeyHash(pkScript, opcode.OP_SSTX)
     if pkHash:
-        return StakeSubmissionTy, pubKeyHashToAddrs(pkHash, chainParams), 1
+        return StakeSubmissionTy, pubKeyHashToAddrs(pkHash, netParams), 1
     scriptHash = extractStakeScriptHash(pkScript, opcode.OP_SSTX)
     if scriptHash:
-        return StakeSubmissionTy, scriptHashToAddrs(scriptHash, chainParams), 1
+        return StakeSubmissionTy, scriptHashToAddrs(scriptHash, netParams), 1
 
     # Check for stake generation script.  Only stake-generation-tagged
     # pay-to-pubkey-hash and pay-to-script-hash are allowed.
     pkHash = extractStakePubKeyHash(pkScript, opcode.OP_SSGEN)
     if pkHash:
-        return StakeGenTy, pubKeyHashToAddrs(pkHash, chainParams), 1
+        return StakeGenTy, pubKeyHashToAddrs(pkHash, netParams), 1
     scriptHash = extractStakeScriptHash(pkScript, opcode.OP_SSGEN)
     if scriptHash:
-        return StakeGenTy, scriptHashToAddrs(scriptHash, chainParams), 1
+        return StakeGenTy, scriptHashToAddrs(scriptHash, netParams), 1
 
     # Check for stake revocation script.  Only stake-revocation-tagged
     # pay-to-pubkey-hash and pay-to-script-hash are allowed.
     pkHash = extractStakePubKeyHash(pkScript, opcode.OP_SSRTX)
     if pkHash:
-        return StakeRevocationTy, pubKeyHashToAddrs(pkHash, chainParams), 1
+        return StakeRevocationTy, pubKeyHashToAddrs(pkHash, netParams), 1
     scriptHash = extractStakeScriptHash(pkScript, opcode.OP_SSRTX)
     if scriptHash:
-        return StakeRevocationTy, scriptHashToAddrs(scriptHash, chainParams), 1
+        return StakeRevocationTy, scriptHashToAddrs(scriptHash, netParams), 1
 
     # Check for stake change script.  Only stake-change-tagged
     # pay-to-pubkey-hash and pay-to-script-hash are allowed.
     pkHash = extractStakePubKeyHash(pkScript, opcode.OP_SSTXCHANGE)
     if pkHash:
-        return StakeSubChangeTy, pubKeyHashToAddrs(pkHash, chainParams), 1
+        return StakeSubChangeTy, pubKeyHashToAddrs(pkHash, netParams), 1
     scriptHash = extractStakeScriptHash(pkScript, opcode.OP_SSTXCHANGE)
     if scriptHash:
-        return StakeSubChangeTy, scriptHashToAddrs(scriptHash, chainParams), 1
+        return StakeSubChangeTy, scriptHashToAddrs(scriptHash, netParams), 1
+
+    # Check for null data script.
+    if isNullDataScript(version, pkScript):
+        # Null data transactions have no addresses or required signatures.
+        return NullDataTy, [], 0
 
-    # EVERYTHING AFTER TIHS IS UN-IMPLEMENTED
```

Comment:
`s/TIHS/THIS`, no dash needed, and why uppercase?

Reply:
cause yelling